src/components/general/SearchableDropdown/index.tsx

```diff
@@ -208,7 +208,11 @@ const SearchableDropdown = ({
                         placeholder={placeholder || 'Type to search...'}
                         label={label}
                         icon={<DropdownArrow cursor="pointer" isOpen={isOpen} />}
-                        onIconAction={() => isOpen && setIsOpen(false)}
+                        onIconAction={(e) => {
+                            e.stopPropagation();
+                            setIsOpen(!isOpen);
+                            ref.current && ref.current.click && ref.current.click();
```

Comment:
There's a problem with `onIconAction` and `onClick` because a click on the icon will also cause the `onClick` to fire off which results in not being able to close the dropdown when clicking the icon. 
This might be a hacky solution, but by stopping event bubbling from the icon, the `onClick` will not run, however stopping event bubbling will also cause a problem with the click outside logic inside `Dropdown`. It can be worked around by calling on the `click()` method from the ref from `useDropdownController`-hook.